### ISSUE TITLE
[PROGRESS] Grace-Hephzibah

### ISSUE THREAD

**Issue by Grace-Hephzibah** [Codezilla-Club/30-Days-of-Code#51]
## Participant Details
**Name/Nickname:** Grace
**Pronouns:** she/her
**Domain:** Machine Learning 

I am AI and ML Enthusiast and I hope to learn as much as possible on this track 

## Milestones
- [x] Code for 30 days

- [x] Explore the mathematical part related to ML
- [x] Become more proficient using Streamlit
- [x] Build at least 1 data model every week - (around 5 data models this month-end)

## Checklist
- [x] I've read the Code of Conduct, Rules and Guidelines
- [x] I've named this Issue as indicated
- [x] I've been assigned a mentor
- [x] Acknowledgement of this post 
## Github Repo Link:
Link: <a href = "https://github.com/Grace-Hephzibah/30-Days-Of-ML"> 30-Days-Of-ML </a>

**Comment by Grace-Hephzibah** [comment]
Day 1

## What I Learned Today
- [x] Object-Oriented Programming in Python
- [x] Setting up and using Anaconda Environment
- [x] Basic Markdown
- [x] Week 1 Project 

Week 1 works mentioned on the Road Map Done  

## What I'll Learn Tomorrow
- Explore regression and related mathematical concepts together with it

**Comment by Grace-Hephzibah** [comment]
Day 2

## What I Learned Today
- [x] Linear Regression using OLS Method 
- [x] Multiple Regression using OLS Method 
- [x] Understanding the usage of scatter plot and line plot 
- [x] Understanding the usage and elimination requirement of unnecessary data   

## What I'll Learn Tomorrow
- Same which I learnt today but using the Sklearn package.

**Comment by Grace-Hephzibah** [comment]
Day 3

## What I Learned Today
- Linear Regression using Sklearn 
- Multiple Regression using Sklearn 
- Underfitting and Overfitting Model
- Train - Test - Split usage

## What I'll Learn Tomorrow
- Logistic Regression

**Comment by Grace-Hephzibah** [comment]
Day 4

## What I Learned Today
- Logistic Regression uses and its non-linearity 
- Binary Prediction and its maths
- Deep analysis on MLE, Log-Likelihood, LL-Null, LLR, and many more
- Simple Logistic Regression 
- Little complicated Logistic Regression 
- Confusion matrix 

## What I'll Learn Tomorrow
- Streamlit and usage of its various components

**Comment by Grace-Hephzibah** [comment]
Day 5

## What I Learned Today
- Covered Streamlit 
- Various text formats and it's rendering style
- Various types of displaying data like a dictionary, data frame, etc
- Displaying Plots and Media 
- Basic Widgets
- Bar, Progress and Status 
- Layouts and Color Schemes 
- Caching in Streamlit  

Resource: <a href = "https://www.youtube.com/playlist?list=PLuU3eVwK0I9PT48ZBYAHdKPFazhXg76h5"> Streamlit Playlist </a>

## What I'll Learn Tomorrow
- K-clustering

**Comment by Grace-Hephzibah** [comment]
Day 6

## What I Learned Today
- General overview of k clustering
- Theoretical examples to understand the topic 

## What I'll Learn Tomorrow
- Code implementation of k-clustering

**Comment by Akshu-on-github** [comment]
Glad to see the progress @Grace-Hephzibah! Thanks for sharing the YT resource - Could you raise an 'Add Resource' Issue? It would be an extra green square for you! 🟩 
If possible, we'd love to see the code that you've worked on - but no pressure 😄

**Comment by Grace-Hephzibah** [comment]
The github repo is linked in the first comment of this issue 
Link:
https://github.com/Grace-Hephzibah/30-Days-Of-ML

I will raise a request to add that as a resource. 😊✨

I am also sharing my progress on twitter just to stay accountable. I really want to push myself. I am tagging Codezilla. 

Thanks for the reply👍🏼

**Comment by Grace-Hephzibah** [comment]
Day 7

## What I Learned Today
- Explored more K Clustering 
- Explored the elbow method
- Understood the idea behind gathering information for every cluster

## What I'll Learn Tomorrow
- Code implementation of K-Clustering 😅

**Comment by Grace-Hephzibah** [comment]
Day 8

## What I Learned Today
- Implemented K-Clustering 
- Implemented the Elbow Method 

## What I'll Learn Tomorrow
- K-Clustering on the Iris Dataset

**Comment by Grace-Hephzibah** [comment]
Day 9

## What I Learned Today
- K-Clustering revision 
- Introduction to Neural Network
- Basics and methods to train a model 

## What I'll Learn Tomorrow
- Continuation of Neural Network

**Comment by Grace-Hephzibah** [comment]
Day 10

## What I Learned Today
- Started working with streamlit with little complexity 

## What I'll Learn Tomorrow
- Continuation of Neural Networks

**Comment by Grace-Hephzibah** [comment]
Day 11

## What I Learned Today
- Joining in elements for tougher and complex views 

## What I'll Learn Tomorrow
- Continuation of Neural Networks

**Comment by Grace-Hephzibah** [comment]
Day 12

## What I Learned Today
- Full revision from first to last. 
- Made notes to some parts that I missed

## What I'll Learn Tomorrow
- Continue Neural Networks

**Comment by Grace-Hephzibah** [comment]
Day 13

## What I Learned Today
- Data processing revision + extra insights 
- Linear model under deep learning (just started)

## What I'll Learn Tomorrow
- Continue with the linear model

**Comment by Grace-Hephzibah** [comment]
Day 14

## What I Learned Today
- A theoretical approach to various types in Linear Model in Neural Networks 

## What I'll Learn Tomorrow
- More on Linear Model

**Comment by Grace-Hephzibah** [comment]
Day 15

## What I Learned Today
- Graphical representation of simple neural networks 
- Objective function. 

## What I'll Learn Tomorrow
- More on the theoretical part on simple neural networks.

**Comment by Grace-Hephzibah** [comment]
Day 16

## What I Learned Today

- Explored L2-norm loss and cross-entropy under the objective function
- Explored Parameter gradient descent under the optimization algorithm

## What I'll Learn Tomorrow
- Revision on all the topics so far covered on neural networks 
- Start coding on this 😅

**Comment by Grace-Hephzibah** [comment]
Day 17

## What I Learned Today
- Full revision of neural networks 

## What I'll Learn Tomorrow
- Coding!

**Comment by Grace-Hephzibah** [comment]
Day 18

## What I Learned Today
- Coding lectures on the linear model

## What I'll Learn Tomorrow
- Continuing the same tomorrow

**Comment by Grace-Hephzibah** [comment]
Day 19

## What I Learned Today
- Completed the coding lectures today 

## What I'll Learn Tomorrow
- Practice sessions on the same

**Comment by Grace-Hephzibah** [comment]
Day 20

## What I Learned Today
- Practiced linear model 

## What I'll Learn Tomorrow
- Starting with TensorFlow

**Comment by Grace-Hephzibah** [comment]
Day 21

## What I Learned Today
- Installed TensorFlow 
- Set up a virtual environment 

## What I'll Learn Tomorrow
- Explore TensorFlow library (overview)

**Comment by Grace-Hephzibah** [comment]
Day 22

## What I Learned Today
- TensorFlow Outline and Comparison with Other Libraries
- TensorFlow 1 vs TensorFlow 2
- Types of File Formats Supporting TensorFlow
- Outlining the Model with TensorFlow 2

## What I'll Learn Tomorrow
- Coding Practice on the same

**Comment by Grace-Hephzibah** [comment]
Day 23

## What I Learned Today
- Customizing models in TensorFlow

## What I'll Learn Tomorrow
- Coding 😅 (the same)

**Comment by Grace-Hephzibah** [comment]
Day 24

## What I Learned Today
- Coded out the same example which I used in NumPy for the linear model in neural networks

## What I'll Learn Tomorrow
- More on Deep Learning

**Comment by Grace-Hephzibah** [comment]
Day 25

## What I Learned Today
Explored
- Layer
- Deep Net
- The purpose of non-linearity
- Activation functions 
- Softmax activation function
- Back Propagation

## What I'll Learn Tomorrow
- Continuation on theory

**Comment by Grace-Hephzibah** [comment]
Day 26

## What I Learned Today
Explored 
- Underfitting
- Overfitting 
- Validation

## What I'll Learn Tomorrow
- More on validation

**Comment by Grace-Hephzibah** [comment]
Day 27

## What I Learned Today
Explored
- N-Fold cross-validation
- Early stopping
- Initialization
- Xavier Initialization

## What I'll Learn Tomorrow
- More on gradient descent

**Comment by Grace-Hephzibah** [comment]
Day 28 

## What I Learned Today
Explored
- Stochastic Gradient Descent
- Drawbacks with Gradient Descent
- Momentum
- Learning Rate Schedules
- Adaptive Learning Rate Schedules (AdaGrad and RMSprop )
- Adam (Adaptive Moment Estimation)

## What I'll Learn Tomorrow
- Data Preprocessing

**Comment by Grace-Hephzibah** [comment]
Day 29

## What I Learned Today
- Data Preprocessing

## What I'll Learn Tomorrow
- More on data preprocessing

**Comment by Grace-Hephzibah** [comment]
Day 30

## What I Learned Today
Explored 
- Data preprocessing
- Standardization
- Preprocessing categorical data
- Binary encoding
- One-hot encoding

## What I'll Learn Tomorrow
- Challenge completed ✨
